app/src/main/java/com/github/se/signify/ui/screens/home/ExerciseScreenEasy.kt

```diff
@@ -200,42 +160,131 @@ fun WordLayer(words: List<String>, currentWordIndex: Int, currentLetterIndex: In
       }
 }
 
-@Composable
-fun SuccessButton(
+/**
+ * Function to handle the gesture matching logic.
+ *
+ * @param detectedGesture The gesture detected by the hand landmark model.
+ * @param currentLetter The current letter to be matched.
+ * @param currentLetterIndex The index of the current letter in the current word.
+ * @param currentWordIndex The index of the current word in the list.
+ * @param wordsList The list of words used in the exercise.
+ * @param onNextLetter Callback to move to the next letter in the current word.
+ * @param onNextWord Callback to move to the next word in the list.
+ * @param onAllWordsComplete Callback to handle when all words are completed.
+ */
+fun handleGestureMatching(
+    detectedGesture: String,
+    currentLetter: Char,
     currentLetterIndex: Int,
     currentWordIndex: Int,
-    words: List<String>,
+    wordsList: List<String>,
     onNextLetter: (Int) -> Unit,
     onNextWord: (Int) -> Unit,
-    onAllWordsComplete: () -> Unit,
-    modifier: Modifier = Modifier
+    onAllWordsComplete: () -> Unit
 ) {
-  Box(modifier = modifier.fillMaxSize().padding(30.dp), contentAlignment = Alignment.BottomCenter) {
-    Button(
-        onClick = {
-          onSuccess(
-              currentLetterIndex = currentLetterIndex,
-              currentWordIndex = currentWordIndex,
-              words = words,
-              onNextLetter = onNextLetter,
-              onNextWord = onNextWord,
-              onAllWordsComplete = onAllWordsComplete)
-        },
-        modifier = Modifier.width(150.dp).height(50.dp).testTag("Success")) {
-          Text(text = "Success")
-        }
+  if (detectedGesture.equals(currentLetter.uppercase())) {
+    onSuccess(
+        currentLetterIndex = currentLetterIndex,
+        currentWordIndex = currentWordIndex,
+        words = wordsList,
+        onNextLetter = onNextLetter,
+        onNextWord = onNextWord,
+        onAllWordsComplete = onAllWordsComplete)
+  } else {
+    Log.d(
+        "ExerciseScreenEasy",
+        "Detected gesture ($detectedGesture) does not match the current letter ($currentLetter)")
   }
 }
 
 @Composable
-fun CameraPlaceholder(modifier: Modifier = Modifier) {
-  Box(
-      modifier =
-          modifier
-              .fillMaxWidth()
-              .padding(16.dp)
-              .height(350.dp)
-              .background(colorResource(R.color.black), shape = RoundedCornerShape(16.dp))
-              .border(2.dp, colorResource(R.color.white), shape = RoundedCornerShape(16.dp)),
-      contentAlignment = Alignment.Center) {}
+fun ExerciseScreenCommon(
```

Comment:
The function ExerciseScreenCommon() could have been created in Utils.kt as it is a common element used in different screen like ExerciseScreenEasy and ExerciseScreenHard.